### PR TITLE
lookup array schemas

### PR DESCRIPTION
**Pull request by marina-mosti**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -22,14 +22,16 @@ export default function LookupPlugin ({ mapComponents = {}, mapProps = {} }) {
   }
 }
 
+export const loopElementsInSchema = (schema, fn) => schema.map(row => row.map(el => fn(el)))
```
Comment (shentao) [inline src/index.js:25]: I’d say it should be `mapFieldsInSchema`. A loop would use a `forEach`.

File: src/index.js [hunk 2]
```diff
@@ -22,14 +22,16 @@ export default function LookupPlugin ({ mapComponents = {}, mapProps = {} }) {
...
+
 /**
  * Remap components in a schema
  * @param {Array} schema - The schema
  * @param {Object|Function} mapComponents
 * @returns {Array}
  */
 const mapComps = (schema, mapComponents) => {
-  return schema.map(el => {
+  return schema.map(row => row.map(el => {
```
Comment (shentao) [inline src/index.js:34]: I think this could use the function that you created above `loopElementsInSchema`.

File: src/index.js [hunk 3]
```diff
@@ -63,12 +67,10 @@ const mapProperties = (schema, mapProps) => {
   }
 
   let schemaCopy
-  if (typeof mapProps === 'object') {
-    for (const prop in mapProps) {
-      schemaCopy = schema.map(el => {
-        return replacePropInElement(el, prop, mapProps[prop])
-      })
-    }
+  for (const prop in mapProps) {
```
Comment (shentao) [inline src/index.js:70]: Truth be told, all those for in loops are extremely hard for me to follow. Could we break it up into `.map` and named map functions? I think this should make it easier to follow and understand later on.

File: src/index.js (hunk 2 above)
Comment (marina-mosti) [reply src/index.js:34]: Good catch, this one slipped

**Pull request merged**